### DISCUSSION TITLE
How to format resource pack that overwrites custom entities with .JEM model? Answer greatly appreciated.

### DISCUSSION THREAD

**Discussion by ParcySnippet** [Luke100000/minecraft-comes-alive#983] [Q&A] [answered]
Hi, I think a few questions similar to this have been posted before, but none of them were answered. I would love to know if it is possible to overwrite the villager's appearance with my own very similar player shaped model that only adds a nose to the head in a resource pack. 

Can this be done while still keeping all the eye and hair texture layers usable in character creation along with the MCA villager's default animations not being overwritten? I have already tried formatting it in many different ways, along with using the name of what I think is the MCA model as the name of the .jem, which is: VillagerEntityModelMCA. If it is possible, the information I would need is how to format such a resource pack. I know that if the model is replaced, the only thing I should need to edit are the texture skin layers in the resource pack to cover the new model, which shouldn't be hard.  

I know I could technically change the villager's appearance by enabling the default squidward models and then using a Resource pack on top of the vanilla villagers, but I would love to be able to keep the character creation customization while still using a custom model.

I would really appreciate an answer, even if it is just an explanation as to why it is impossible, or if it still could theoretically be done, even if no one is sure how. Thank you for reading.

**Comment by Luke100000** [comment]
Right now, the player model is constructed with Java. You can extend that all you want but a resourcepack probably won't do it.

**Reply by ParcySnippet** [reply]
Thank you for answering. A few questions I have then: 

Would only one jar would have to be edited to add that feature? If so, which one? And out of curiosity, since I'm not a coder, can the player model be changed with a visual aid? I browsed through the .jar and saw a few components labeled stuff like "head" or "arm" but it was nothing close to readable like the code of the json of a .jem model. 

So would there be a place I could insert and replace the model code with said .json?  Sorry if these are dumb questions, but just knowing whether or not there is no way I could execute this without learning a lot of coding will probably put me at ease from trying to waste my time throwing stuff at the wall. Thank you for your time.

**Reply by Luke100000** [reply, marked as answer]
You would need to download this repo and build your own version, and there is no Json used at all. Without intermediate Java knowledge, you won't be able to do that.

**Reply by ParcySnippet** [reply]
Thank you, at least I can be sure now. I will continue making my resource pack without the extra model I wanted and put that idea to rest.
